### PR TITLE
handle cleanup for delete operations to Solr, fixing up allow-mult=true

### PR DESCRIPTION
**Pull request by zeeshanlakhani**
- rewrite cleanup method into cleanup and cleanup_for_sibs, the latter for handling allow-mult=true objects' sibling reconciliation and tombstones
- make yz_siblings r_t use delete_mode=keep so that we can test how yz handles cleanup for allow-mult=true reconciliation/delete

### CODE REVIEW

File: riak_test/yz_siblings.erl [hunk 1]
```diff
@@ -88,7 +137,8 @@ reconcile_sibs(Cluster, HP, Index, Bucket, EncKey) ->
 delete_key(Cluster, HP, Index, Bucket, EncKey) ->
     lager:info("Delete the key"),
     URL = bucket_url(HP, Bucket, EncKey),
-    {ok, "200", RH, _} = ibrowse:send_req(URL, [], get, [], []),
+    {ok, Status, RH, _} = ibrowse:send_req(URL, [], get, [], []),
+    ?assert(Status == "200" orelse Status == "300"),
```
Comment (fadushin) [inline riak_test/yz_siblings.erl:141]: I am curious why checking for 300 is now required, but I am not completely up on the Riak HTTP API.  Will Riak send back a 300 after a partition heal?

Comment (zeeshanlakhani) [inline riak_test/yz_siblings.erl:141]: cause we have siblings, so we may get multiple pieces content.

**Comment by fadushin** [comment]
+1 cd4badc

**Comment by zeeshanlakhani** [comment]
@borshop merge

**Pull request merged**